### PR TITLE
Goa 2443 use individual presets

### PR DESCRIPTION
**Pull request by ljgarcia**

### CODE REVIEW

File: app/advancedfilters/taxonFilter.js [hunk 1]
```diff
@@ -31,29 +29,56 @@ app.controller('taxonFilter', function($scope, hardCodedDataService,
...
+    if (taxaIds.length !== 0) {
+      var taxonomyPromise = taxonomyService.getTaxa(taxaIds);
+      taxonomyPromise.then(function (multipleTaxa) {
+        angular.forEach(multipleTaxa.data.taxonomies, function (taxon) {
+          $scope.taxaMapping[taxon.taxonomyId] = taxon;
+        });
+        angular.forEach(multipleTaxa.data.errors, function (error) {
+          delete $scope.taxaIds[error.requestedId];
+        });
+        if (withRedirection === true) {
```
Comment (xwatkins) [inline app/advancedfilters/taxonFilter.js]: This and the recursion make the code difficult to read. getTaxaInfo(taxIds) should just do _one_ thing: retrieve data and populate the objects. It is then the responsibility of the function which made the initial call to make another call to get redirected ids if any.

File: app/advancedfilters/taxonFilter.js [hunk 2]
```diff
@@ -1,27 +1,25 @@
 app.controller('taxonFilter', function($scope, hardCodedDataService,
-  stringService, validationService){
+  stringService, validationService, presetsService, taxonomyService){
 
-  $scope.taxonIds = [];
+  $scope.taxaIds = {};
+  $scope.taxaMapping = {};
```
Comment (ljgarcia) [inline app/advancedfilters/taxonFilter.js]: Agree, we will have only $scope.taxa now.

**Comment by ljgarcia** [comment]
Gene product was the only one using an array rather than an object for preset results. It was changed to an object in Use preset service and moves to {} as in similar pages (cf91759).

File: app/advancedfilters/geneProductFilter.js [hunk 3]
```diff
@@ -1,43 +1,45 @@
 app.controller('geneProductFilter', function ($scope, stringService,
-  validationService, hardCodedDataService) {
+  validationService, presetsService) {
 
   $scope.gpIds = [];
```
Comment (xwatkins) [inline app/advancedfilters/geneProductFilter.js]: Please also use a map for gpIds

File: app/advancedfilters/taxonFilter.js [hunk 4]
```diff
@@ -1,27 +1,26 @@
...
-            'checked': true
-          });
-        }
+    presetsService.getPresetsTaxa().then(function(resp){
+      var checked = [];
+      if($scope.query.taxonId) {
+        checked = checked.concat($scope.query.taxonId.split(','))
+      }
+
+      var ids = [];
```
Comment (xwatkins) [inline app/advancedfilters/taxonFilter.js]: Not needed, just use $scope.taxa keys (undercore provide a method for this)

File: app/advancedfilters/taxonFilter.js [hunk 5]
```diff
@@ -30,32 +29,84 @@ app.controller('taxonFilter', function($scope, hardCodedDataService,
...
+      cleanTaxaInformation(tempTaxa, data);
+    });
     $scope.taxonTextArea = '';
   };
 
+  var getQuery = function() {
+    return _.pluck(_.filter($scope.taxa, 'checked'), 'name');
+  };
+
+  var getTaxaInfo = function (taxaIds) {
```
Comment (xwatkins) [inline app/advancedfilters/taxonFilter.js]: This is redundant as is just creating a promise to return a promise...

**Pull request merged**